graylog2-web-interface/src/components/sidecars/failure-tracking/SidecarFailureTrackingList.tsx

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { Button, Col, Row, Table } from 'components/bootstrap';
+import { PaginatedList, NoSearchResult, NoEntitiesExist, SortIcon } from 'components/common';
+import SidecarSearchForm from 'components/sidecars/common/SidecarSearchForm';
+
+import SidecarFailureTrackingRows from './SidecarFailureTrackingRows';
+
+import VerboseMessageModal from '../sidecars/VerboseMessageModal';
+import type { Collector, PaginationInfo, SidecarSummary } from '../types';
+
+const StyledSortIcon = styled(SortIcon)`
+  && {
+    width: 12px;
+    margin-left: 5px;
+    line-height: 1;
+  }
+`;
+
+const StandardWidthCol = styled.col`
+  width: 9%;
+`;
+
+const ErrorMessageCol = styled.col`
+  width: 14%;
+`;
+
+const VerboseMessageCol = styled.col`
+  width: 50%;
+`;
+
+type Props = {
+  sidecars: SidecarSummary[],
+  collectors: Collector[],
+  pagination: PaginationInfo,
+  query: string,
+  sort: { field: string, order: string },
+  onlyActive: boolean,
+  pageSizes: number[],
+  onPageChange: (page: number, pageSize: number) => void,
+  onQueryChange: (query: string) => void,
+  onSortChange: (sortField: string) => void,
+  toggleShowInactive: () => void,
+}
+
+const SidecarFailureTrackingList = ({
+  sidecars,
+  collectors,
+  pagination,
+  query,
+  sort,
+  onlyActive,
+  pageSizes,
+  onPageChange,
+  onQueryChange,
+  onSortChange,
+  toggleShowInactive,
+}: Props) => {
+  const [collectorDetailsToShow, setCollectorDetailsToShow] = useState<{ name: string, verbose_message: string }|null>(null);
+
+  const formatSidecarList = (sidecarRows: React.ReactNode[]) => {
+    const sidecarCollection = {
```

Comment:
In this case, I find the naming confusing. 
```suggestion
    const columns = {
```

or maybe `columnsMap`